packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -166,7 +166,7 @@ function newView(
   const customTimings: ViewCustomTimings = {}
   let documentVersion = 0
   let endClocks: ClocksState | undefined
-  let location: Location = { ...initialLocation }
+  let location: ReadonlyLocation = cloneLocation(initialLocation)
```

Comment:
What was the issue with the previous approach?
Could we keep it and directly use `Readonly<Location>`
it would avoid to explicitly list properties

Reply:
The issue I tried to avoid was exposing `location` methods, because they can't be used (they throw an error):

```js
({ ...location }).reload()
// Uncaught TypeError: Illegal invocation
```

Reply:
if we make clear that this is not the actual location, throwing an error seems an OK behavior to me

Reply:
Alright, I changed it back